processing/src/main/java/org/apache/druid/guice/LifecycleForkJoinPool.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.guice;
+
+
+import org.apache.druid.java.util.common.lifecycle.LifecycleStop;
+import org.apache.druid.java.util.common.logger.Logger;
+
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.TimeUnit;
+
+public class LifecycleForkJoinPool extends ForkJoinPool
```

Comment:
This seems to be the cause of mysterious java 9+ compilation issues causing CI failures (my favorite part is how the compiler error doesn't at all specify what is causing the error). `ForkJoinPool` has the `@Contended` annotation, which from javadocs looks like it leaks to subclasses I guess?
```
 * <p>The class level {@code @Contended} annotation is not inherited and has
 * no effect on the fields declared in any sub-classes. The effects of all
 * {@code @Contended} annotations, however, remain in force for all
 * subclass instances, providing isolation of all the defined contention
 * groups. Contention group tags are not inherited, and the same tag used
 * in a superclass and subclass, represent distinct contention groups.
```

This might be fixed by something like:

```
<compilerArgs>
  <arg>--add-exports</arg>
  <arg>java.base/jdk.internal.misc=ALL-UNNAMED</arg>
</compilerArgs>
```

added to jdk9+ profile, but would also require removing all source/target directives into a jdk8 profile I think, since:

```
 error: option --add-exports not allowed with target 8
```
happens otherwise.

Not sure what is the best thing to do here. it could probably be worked around by I guess having a lifecycle'd module that wraps a `ForkJoinPool` for the provider instead of extending `ForkJoinPool`, since just using `ForkJoinPool` doesn't seem to cause this error in jdk9+, but I'm not sure if that is the better solution, or if we should try to add the compiler args.